pyne/xs/models.pyx

```diff
@@ -160,8 +160,31 @@ def phi_g(E_g, E_n, phi_n):
     phi_g = np.dot(pem, phi_n)
     return phi_g
 
+def number_density(mat, density, a_mass):
+    """Builds the weights used during the self shielding calculations. 
 
-def group_collapse(sigma_n, phi_n, phi_g=None, partial_energies=None, E_g=None, E_n=None):
+    Parameters
+    ----------
+    mat : dictionary of isotope to concentation  
+        The mass based isotopic composition of the material for which self
+        shielding is being calculated. 
+    density : float
+        the density of the material
+    a_mass : float
+        the atomic mass of the material
+    
+    Returns
+    -------
+    num_den : dictionary
+        dictionary of isotope to number density
+    """
+    num_den_mat = 6.0221409E23 * density / a_mass
+    num_den = {}
+    for iso in mat:
+        num_den[iso] = mat[iso] * num_den_mat[0]
+    return num_den
```

Comment:
This whole function seems like you are reinventing the whell: http://pyne.io/pyapi/material.html#pyne.material.Material.number_density